### PR TITLE
Gamemode system

### PR DESCRIPTION
**Pull request by DMP9-MC**
Does what it says on the tin -- required for #17

### CODE REVIEW

File: prison-core/src/main/java/tech/mcprison/prison/internal/Player.java [hunk 1]
```diff
@@ -77,7 +78,9 @@
      */
     void setScoreboard(Scoreboard scoreboard);
 
+    Gamemode getGamemode();
```
Comment (SirFaizdat) [inline prison-core/src/main/java/tech/mcprison/prison/internal/Player.java]: Needs documentation

Comment (SirFaizdat) [inline prison-core/src/main/java/tech/mcprison/prison/internal/Player.java]: Needs documentation

File: prison-core/src/main/java/tech/mcprison/prison/internal/Player.java [hunk 2]
```diff
@@ -77,7 +78,9 @@
...
 
+    void setGamemode(Gamemode gamemode);
```
Comment (SirFaizdat) [inline prison-core/src/main/java/tech/mcprison/prison/internal/Player.java]: Needs documentation

File: prison-core/src/main/java/tech/mcprison/prison/util/Gamemode.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package tech.mcprison.prison.util;
+
+
+public enum Gamemode {
```
Comment (SirFaizdat) [inline prison-core/src/main/java/tech/mcprison/prison/util/Gamemode.java]: Needs documentation

File: prison-core/src/main/java/tech/mcprison/prison/internal/Player.java (hunk 2 above)
Comment (DMP9-MC) [inline prison-core/src/main/java/tech/mcprison/prison/internal/Player.java]: While I do agree with this and will be committing to implement, half of prison-core doesn't have documentation. I'll be happy to write the missing documentation, I just find it a bit hypocritical.

**Pull request merged**